.github/workflows/build_and_deploy.yml

```diff
@@ -130,11 +129,20 @@ jobs:
           fi
         shell: bash
 
-      # Use this one for testing: python -m twine upload --repository-url https://test.pypi.org/legacy/ --skip-existing dist/pmdarima-*
       - name: Deploying to PyPI
         # Only deploy on tags and when VERSION file created
         if: github.event_name == 'tag' && success() && steps.version_check.version_exists == 'true'
-        run: python -m twine upload --skip-existing dist/pmdarima-*
+        run: |
+          # Check our VERSION. Basically, if it contains letters, it is a pre-release. Otherwise,
+          # it has to match X.Y or X.Y.Z
+          if [[ $(cat ${GITHUB_WORKSPACE}/pmdarima/VERSION) =~ '^[0-9]+\.[0-9]+\.?[0-9]*?[a-zA-Z]+[0-9]*$' ]]; then
```

Comment:
Currently only something like `1.0.3rc1` is caught by this. What if someone tags with something like `1.0.3-rc1`? Should we make more flexible like:

`^[0-9]+\.[0-9]+\.?[0-9]*?-?[a-zA-Z]+[0-9]*$`

Reply:
Or does it make sense to very strictly enforce the current supported regex?

Reply:
Also our `get_tag.py` supports the potential of a tag leading with `v`. Should we make the regex start with `^v?`?

Reply:
RE: echoing, that is easy enough, I can add that.

RE: Regex, I am open to making it more flexible, so if you would like it to support hyphens, I am all for that. Technically, there is a published way these should look ([PEP 440](https://www.python.org/dev/peps/pep-0440/#public-version-identifiers)), which I have mostly covered, but I can add support for the `post` and `dev` tags

RE: Leading V, I don't think we need to add that on Azure or Github. The reason being, we are reading the actual `VERSION` file, which translates `v1.5.3` to `1.5.3`. On the Circle job, since we use the **tag** and not the `VERSION` file, you can see we _do_ support the leading V